plugins/azure-sites-backend/src/service/router.ts

```diff
@@ -19,19 +19,32 @@ import express from 'express';
 import Router from 'express-promise-router';
 import { Logger } from 'winston';
 
+import { stringifyEntityRef } from '@backstage/catalog-model';
+import { NotAllowedError } from '@backstage/errors';
+import { getBearerTokenFromAuthorizationHeader } from '@backstage/plugin-auth-node';
+import {
+  PermissionEvaluator,
+  AuthorizeResult,
+} from '@backstage/plugin-permission-common';
+import {
+  azureSitesActionPermission,
+  AZURE_WEB_SITE_NAME_ANNOTATION,
+} from '@backstage/plugin-azure-sites-common';
+
 import { AzureSitesApi } from '../api';
 
 /** @public */
 export interface RouterOptions {
   logger: Logger;
   azureSitesApi: AzureSitesApi;
+  permissions: PermissionEvaluator;
```

Comment:
this is a breaking change, as adopters need to manually pass `permissions`. Perhaps we could make it optional
```suggestion
  permissions?: PermissionEvaluator;
```

Reply:
If we make it optional any one can do start and stop operation if the adopter didn't pass the permission. WDYT @vinzscam

Reply:
yeah maybe we could throw a warning at startup if we detect that `permissions` hasn't been passed